cmd/dep/gopath_scanner.go

```diff
@@ -52,10 +52,9 @@ func (g *gopathScanner) InitializeRootManifestAndLock(rootM *dep.Manifest, rootL
 		return err
 	}
 
-	g.origM = &dep.Manifest{
-		Constraints: g.pd.constraints,
-		Ovr:         make(gps.ProjectConstraints),
-	}
+	g.origM = dep.NewManifest()
+	g.origM.Constraints = g.pd.constraints
```

Comment:
Won't it be `O(N+1)` manifests/locks for `N` imported projects (and self)?
It's still not a big deal (at least not for now). My WIP cache branch messes with these a bit (since locks and manifests are serialized/deserialized), so if it worsens the situation then I can revisit it there.